dynamiq/memory/backends/pinecone.py

```diff
@@ -16,45 +17,20 @@ class PineconeError(Exception):
 
 
 class Pinecone(MemoryBackend):
-    name = "Pinecone"
-
-    def __init__(
-        self,
-        connection: PineconeConnection,
-        embedder: BaseEmbedder,
-        index_type: PineconeIndexType,
-        index_name: str = "conversations",
-        namespace: str = "default",
-        cloud: str | None = None,
-        region: str | None = None,
-        environment: str | None = None,
-        pod_type: str | None = None,
-        pods: int = 1,
-    ):
-        """Initializes the Pinecone memory storage."""
-        self.connection = connection
-        self.index_name = index_name
-        self.embedder = embedder
-        self.namespace = namespace
+    """Pinecone memory backend implementation."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
 
+    name: str = "Pinecone"
+    connection: PineconeConnection
+    embedder: BaseEmbedder
+    vector_store: PineconeVectorStore
+
+    def model_post_init(self, __context) -> None:
+        """Verify connection after model initialization."""
         try:
-            self.vector_store = PineconeVectorStore(
-                connection=connection,
-                index_name=index_name,
-                namespace=namespace,
-                dimension=embedder.dimensions,
-                create_if_not_exist=True,
-                index_type=index_type,
-                cloud=cloud,
-                region=region,
-                environment=environment,
-                pod_type=pod_type,
-                pods=pods,
-            )
-            # Verify connection
             if not self.vector_store._index:
```

Comment:
Let's make single raise with proper message, instead of raise, catch and extra raise